snarkos/ledger/mod.rs

```diff
@@ -310,17 +315,21 @@ impl<N: Network> Ledger<N> {
                     trace!("Requesting block {height} of {latest_height}");
 
                     // Download the block with an exponential backoff retry policy.
-                    handle_dispatch_error(move || async move {
-                        // Get the URL for the block download.
-                        let block_url = format!("{TARGET_URL}{height}.block");
+                    let client_clone = client.clone();
```

Comment:
this clone is a bit annoying (wasn't able to get around it), but it's only an `Arc` bump